tnt/src/main/java/org/dcsa/conformance/standards/tnt/party/TntPublisher.java

```diff
@@ -258,4 +258,53 @@ private static int compareNodesByField(JsonNode node1, JsonNode node2, String fi
       return 0; // Consider how to handle incomparable types
     }
   }
+
+  protected String generateCursor(ArrayNode events, int limit, String cursorKey) {
+    if (events.size() <= limit) {
+      persistentMap.save(cursorKey, OBJECT_MAPPER.createObjectNode());
+      return null;
+    }
+    JsonNode lastEvent = events.get(limit - 1);
+    String lastEventId = lastEvent.at("/eventID").asText();
+    persistentMap.save(cursorKey, OBJECT_MAPPER.createObjectNode().put("cursor", lastEventId));
+    return lastEventId;
+  }
+
+  private ArrayNode applyCursorLogic(ArrayNode filteredArray, String cursor, String cursorKey, int limit, Map<String, Collection<String>> headers) {
+    // Retrieve cursor from persistentMap if available
+    if (cursor != null) {
```

Comment:
What does is actually mean, when you enter this method and `cursor` is null? Should it return directly, or should it have/behave as a default value?